File: Sources/iStream/Models/PipelineInput.swift
import Foundation

public struct PipelineInput: Equatable, Sendable {
    public var topicName: String
    public var fields: [Field]
    public var partitionKey: String
    public var cardinality: Int
    public var skew: SkewLevel
    public var throughputEventsPerSec: Int
    public var peakMultiplier: Double
    public var avgEventSizeBytes: Int
    public var latencySLOms: Int

    public init(
        topicName: String,
        fields: [Field],
        partitionKey: String,
        cardinality: Int,
        skew: SkewLevel,
        throughputEventsPerSec: Int,
        peakMultiplier: Double,
        avgEventSizeBytes: Int,
        latencySLOms: Int
    ) {
        self.topicName = topicName
        self.fields = fields
        self.partitionKey = partitionKey
        self.cardinality = cardinality
        self.skew = skew
        self.throughputEventsPerSec = throughputEventsPerSec
        self.peakMultiplier = peakMultiplier
        self.avgEventSizeBytes = avgEventSizeBytes
        self.latencySLOms = latencySLOms
    }
}

File: lib/models/pipeline_input.dart
import 'field.dart';
import 'skew_level.dart';

class PipelineInput {
  const PipelineInput({
    required this.topicName,
    required this.fields,
    required this.partitionKey,
    required this.cardinality,
    required this.skew,
    required this.throughputEventsPerSec,
    required this.peakMultiplier,
    required this.avgEventSizeBytes,
    required this.latencySLOms,
  });

  final String topicName;
  final List<Field> fields;
  final String partitionKey;
  final int cardinality;
  final SkewLevel skew;
  final int throughputEventsPerSec;
  final double peakMultiplier;
  final int avgEventSizeBytes;
  final int latencySLOms;
}
